managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -3633,12 +3633,19 @@ public boolean isValidPosition(PositionImpl position) {
             log.debug("IsValid position: {} -- last: {}", position, last);
         }
 
-        if (position.getEntryId() < 0) {
+        if (!ledgers.containsKey(position.getLedgerId())){
             return false;
-        } else if (position.getLedgerId() > last.getLedgerId()) {
+        } else if (position.getEntryId() < 0) {
             return false;
+        } else if (position.getLedgerId() > last.getLedgerId() && position.getLedgerId() != currentLedger.getId()) {
+            return false;
+        } else if (position.getLedgerId() == currentLedger.getId()) {
+            return position.getEntryId() <= currentLedgerEntries;
+        } else if (position.getLedgerId() == currentLedger.getId()) {
+            return position.getEntryId() <= currentLedgerEntries;
         } else if (position.getLedgerId() == last.getLedgerId()) {
-            return position.getEntryId() <= (last.getEntryId() + 1);
+            // Only last ledger can set entry id larger than the last entry.
+            return position.getEntryId() <= last.getEntryId();
```

Comment:
Interesting. Not sure why used `+1` before.

Reply:
Because this logic branch contains two cases: 
- `lastConfirmedPosition,ledger` equals `ml.currentLedger`
- `lastConfirmedPosition,ledger` is the latest ledger which is not empty, but not equal `ml.currentLedger`.  For example: LAC is `4:3`, current ledger is `5` and it is empty.

In this patch, I split this logic branch into two:
- if equals `ml.currentLedger`, do `+1`
- else, do not `+1`